src/services/WaterSchedulingService.ts

```diff
@@ -1,29 +1,38 @@
 import axios from "axios";
+import { GroupedIntervals } from "src/domain/GroupedIntervals";
 import Interval from "src/domain/Interval";
+import { IntervalDto } from "src/domain/IntervalDto";
+import { convertGroupedIntervalsToIntervalDtoArray } from "src/utils/intervalParser";
 
 export async function getSchedule(): Promise<Interval[]> {
-  let schedule;
-
+  let intervalDtos: IntervalDto[];
+  let schedule: Interval[];
   try {
     let url = `http://localhost:3100/api/schedule`;
     const response = await axios.get(url);
-    if (response.status !== 200) return [];
+    if (Math.floor(response.status / 100) !== 2) return [];
```

Comment:
This needs to go away. Axios is handling errors on its own

Reply:
Removed.